File: 0896-monotonic-array/0896-monotonic-array.go
func isMonotonic(nums []int) bool {
    var dir int = 2
    for i := 1; i < len(nums); i++ {
        if nums[i] > nums[i-1] {
            if dir == 0 {
                return false
            } else {
                dir = 1
            }
        } else if nums[i] < nums[i-1] {
            if dir == 1 {
                return false
            } else {
                dir = 0
            }
        }
    }
    return true
}

File: 0896-monotonic-array/0896-monotonic-array.cpp
int speedUp = [] {
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);
    return 0;
}();
 
class Solution {
    public:
    bool isMonotonic(const vector<int>& nums) {
        int prev = nums[0];
        int dir = 0;
        for (const int el : nums) {
            if (el > prev) {
                if (dir == -1) return false;
                dir = 1;
        } else if (el < prev) {
            if (dir == 1) return false;
            dir = -1;
        }
        prev = el;
        }
        return true;
    }
};